File: typescript/docs_code_snippets/embeddings-create.ts
import { z } from "zod";
import { Opper } from "opperai";

const opper = new Opper();

const result = await opper.call("embed-text", {
  input: { text: "The quick brown fox jumps over the lazy dog" },
  input_schema: z.object({ text: z.string() }),
  output_schema: z.object({ embedding: z.array(z.number()) }),
  model: "openai/text-embedding-3-small",
});

console.log(`Dimensions: ${result.data.embedding.length}`);

File: python/docs_code_snippets/embeddings-create.py
from opperai import Opper

opper = Opper()

# --- docs ---
result = opper.call(
    "embed-text",
    input={"text": "The quick brown fox jumps over the lazy dog"},
    input_schema={"type": "object", "properties": {"text": {"type": "string"}}},
    output_schema={
        "type": "object",
        "properties": {"embedding": {"type": "array", "items": {"type": "number"}}},
    },
    model="openai/text-embedding-3-small",
)

print(f"Dimensions: {len(result.data['embedding'])}")
# --- /docs ---
